src/xbot/common/subsystems/ConnectionMonitorSubsystem.java

```diff
@@ -0,0 +1,43 @@
+package xbot.common.subsystems;
+
+import com.google.inject.Singleton;
+import xbot.common.command.BaseSubsystem;
+import xbot.common.logic.Latch;
+import xbot.common.properties.DoubleProperty;
+import xbot.common.properties.XPropertyManager;
+
+import java.util.Observable;
+import java.util.Observer;
+
+@Singleton
+public class ConnectionMonitorSubsystem extends BaseSubsystem implements Observer {
+
+    protected final DoubleProperty timeOut;
+    private final Latch connectionLatch = new Latch(true, Latch.EdgeType.FallingEdge);
+
+    private double lastPacketReceivedTimestamp = -1;
+    private double previousDisconnectionTimestamp = -1;
+
+    public ConnectionMonitorSubsystem(XPropertyManager propertyManager) {
+        log.info("Creating");
+        timeOut = propertyManager.createPersistentProperty("ConnectionMonitor TimeOut Threshold Seconds", 1.0);
+        connectionLatch.addObserver(this);
+    }
+
+    public void setLastPacketReceivedTimestamp(double currentTimestamp) {
+        if (lastPacketReceivedTimestamp != -1) {
```

Comment:
😨 Agh! Never compare floating-point values for equality! Use `< 0` instead here.